parquet/src/main/java/org/apache/iceberg/parquet/ParquetDictionaryRowGroupFilter.java

```diff
@@ -75,6 +86,16 @@ public ParquetDictionaryRowGroupFilter(Schema schema, Expression unbound, boolea
    */
   public boolean shouldRead(MessageType fileSchema, BlockMetaData rowGroup,
                             DictionaryPageReadStore dictionaries) {
+    StructType struct;
+
+    if (nameMapping != null) {
+      MessageType project = ParquetSchemaUtil.pruneColumnsByName(fileSchema, schema, nameMapping);
+      struct = ParquetSchemaUtil.convert(project).asStruct();
+    } else {
+      struct = schema.asStruct();
+    }
+
+    this.expr = Binder.bind(struct, Expressions.rewriteNot(expr), caseSensitive);
```

Comment:
This class doesn't need to know about the name mapping. The mapping should be used to add IDs to the file schema so that most classes don't need to add specific support.

Look at how the other fallback happens using `ParquetSchemaUtil.addFallbackIds`. I think this should mimic that fallback in all cases.

Reply:
The added unit test failed if we only use the ID fallback strategy. How about we keep both the original logic and the name mapping logic?

Reply:
We need to keep both ID assignment strategies, but we don't need to change these filters. Just pass the file schema with assigned IDs into this like before.

Reply:
I changed to pass `typeWithIds`.

Reply:
Why is the name mapping still passed in here?

Reply:
The evaluation build metadata mapping with `BlockMetadata` read from the footer which may contain columns that not in the file schema passed in, so I use name mapping here to filter out them.  The new commit changed to use a try/catch block.

Reply:
The Parquet schema will match exactly because the IDs were added to the file schema. Adding IDs doesn't change the names, so `meta.getPath()` will be the same one used by `fileSchema.getType(...)`.